src/zkey/pkey.c

```diff
@@ -173,6 +201,9 @@ bool is_ep11_aes_key(const u8 *key, size_t key_size)
 	if (ep11key->version != EP11_STRUCT_MAGIC)
 		return false;
 
+	if (memcmp(key + sizeof(struct ep11kblob_header), nulls, sizeof(nulls)) != 0)
```

Comment:
Hm, `sizeof(nulls)` may be wrong, dependent how long sizeof(struct ep11kblob_header) is ....
I would make the `nulls` 32 bytes long and calculate the number of bytes you compare using sizeof(session-field) - sizeof(overlayed-hdr-len).

Also you have `key + sizeof(struct ep11kblob_header)` but `struct ep11keytoken` defines `head` slightly different than `struct ep11kblob_header`. `Head `is only 8 bytes, but `struct ep11kblob_header` is 16 bytes....

In zkey I have recently changed `struct ep11keytoken` to:

```
struct ep11keytoken {
	union {
		u8 session[32];
		struct ep11kblob_header head;
	};
```
Note that this also requires slight adoptions at code places where the head fileds are used, since the field names are slightly different.

Reply:
Hmm ... to keep it simple I would rather prefer to get back to the previous coding:
```
	if (memcmp(key + 16, nulls, 16) != 0)
		return false;
```
This is what the comment says: "Let's require that the 2nd 16 bytes of the session id field are always zero, otherwise consider the key as corrupted."

Reply:
Yea well, probably the best to do so...

Nevertheless, I would think it is good to align the definition of struct ep11keytoken  to how it (now) in in the kernel header:
```
struct ep11keytoken {
	union {
		u8 session[32];
		struct ep11kblob_header head;
	};
```
With that `head` is also 16 bytes, and you could then savely use `key + sizeof(struct ep11kblob_header)`....

Reply:
ok, but let's do this in a separate pr because it would change a struct that is used everywhere in libzpc